django_cte/query.py

```diff
@@ -81,10 +81,26 @@ def generate_sql(cls, connection, query, as_sql):
             ctes.append(cls.TEMPLATE.format(name=qn(cte.name), query=cte_sql))
             params.extend(cte_params)
 
-        # Always use WITH RECURSIVE
-        # https://www.postgresql.org/message-id/13122.1339829536%40sss.pgh.pa.us
-        sql = ["WITH RECURSIVE", ", ".join(ctes)] if ctes else []
+        explain_query = getattr(query, "explain_query", None)
+        sql = []
+        if explain_query:
+            explain_format = getattr(query, "explain_format", None)
+            explain_options = getattr(query, "explain_options", {})
+            sql.append(connection.ops.explain_query_prefix(explain_format, **explain_options))
+            # this needs to get set to False so that the base as_sql() doesn't insert the EXPLAIN statement where it
+            # would end up between the WITH ... clause and the final SELECT
+            query.explain_query = False
+
+        if ctes:
+            # Always use WITH RECURSIVE
+            # https://www.postgresql.org/message-id/13122.1339829536%40sss.pgh.pa.us
+            sql.extend(["WITH RECURSIVE", ", ".join(ctes)])
         base_sql, base_params = as_sql()
+
+        # restore the explain_query value
+        if explain_query is not None:
```

Comment:
```suggestion
        if explain_query:
```

Reply:
I'd specifically checked `is not None` so as to not set an attribute if there was none there previously.

Reply:
`explain_query` will not evaluate to true if the attribute was not previously set (it would be `None` in that case).  However, as written it will unnecessarily set `query.explain_query = explain_query` if `explain_query` evaluates to false and is not `None`.

Reply:
of course 🤦 ... it's been a rough re-entry today, ha